bazel_tools/haskell-c2hs-windows.patch

```diff
@@ -0,0 +1,45 @@
+diff --git a/haskell/c2hs.bzl b/haskell/c2hs.bzl
+index f4f9f42..9aa9a8c 100644
+--- a/haskell/c2hs.bzl
++++ b/haskell/c2hs.bzl
+@@ -39,6 +39,7 @@ def _c2hs_library_impl(ctx):
+     args.add("-C-includeghcversion.h")
+     args.add_all(["-C" + x for x in cc.cpp_flags])
+     args.add_all(["-C" + x for x in cc.include_args])
++    args.add_all(ctx.attr.extra_args)
+ 
+     dep_chi_files = [
+         dep[C2hsLibraryInfo].chi_file
+@@ -68,12 +69,23 @@ def _c2hs_library_impl(ctx):
+             set.to_depset(version_macro_headers),
+         ]),
+         outputs = [hs_file, chi_file],
+-        command = """
++        command =
++        # cpp (called via c2hs) gets very unhappy if the mingw bin dir is
++        # not in PATH so we add it to PATH explicitely.
++        (
++        """
++        export PATH=$PATH:{cc}/..
```

Comment:
Is that a legal path? `{cc}` points to a file. If I try
```
touch foo
ls foo/..
```
I get "not a directory".

Reply:
That is a very good point, confusingly it seems to work …

Reply:
Switched to `paths.dirname(cc)` which seems less stupid.